sdk1/src/test/java/com/amazonaws/services/dynamodbv2/mapper/integration/MissingSignatureFieldsITCase.java

```diff
@@ -0,0 +1,112 @@
+ */
+package com.amazonaws.services.dynamodbv2.mapper.integration;
+
+import static org.testng.Assert.assertEquals;
+
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBAttribute;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBHashKey;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMapper;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMappingException;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBTable;
+import com.amazonaws.services.dynamodbv2.datamodeling.encryption.DoNotTouch;
+import com.amazonaws.services.dynamodbv2.mapper.encryption.StringAttributeTestClass;
+import com.amazonaws.services.dynamodbv2.mapper.encryption.TestDynamoDBMapperFactory;
+import com.amazonaws.services.dynamodbv2.model.AttributeValue;
+import com.amazonaws.services.dynamodbv2.model.PutItemRequest;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+/** Test reading of document which does not contain signature field. */
+public class MissingSignatureFieldsITCase extends DynamoDBMapperCryptoIntegrationTestBase {
+
+  private static final String ORIGINAL_NAME_ATTRIBUTE = "originalName";
+  private static final String STRING_ATTRIBUTE = "stringAttribute";
+  private static final List<Map<String, AttributeValue>> attrs = new LinkedList<>();
+
+  // Test data
+  static {
+    for (int i = 0; i < 5; i++) {
+      Map<String, AttributeValue> attr = new HashMap<String, AttributeValue>();
+      attr.put(KEY_NAME, new AttributeValue().withS("" + startKey++));
+      attr.put(STRING_ATTRIBUTE, new AttributeValue().withS("" + startKey++));
+      attr.put(ORIGINAL_NAME_ATTRIBUTE, new AttributeValue().withS("" + startKey++));
+      attrs.add(attr);
+    }
+  }
+
+  @BeforeClass
+  public static void setUp() throws Exception {
+    DynamoDBMapperCryptoIntegrationTestBase.setUp();
+    for (Map<String, AttributeValue> attr : attrs) {
+      dynamo.putItem(new PutItemRequest(TABLE_NAME, attr));
+    }
+  }
+
+  @Test
+  public void testLoadWithMissingSignatureFields() {
+    DynamoDBMapper util = TestDynamoDBMapperFactory.createDynamoDBMapper(dynamo);
+
+    for (Map<String, AttributeValue> attr : attrs) {
+      AllDoNotTouchTable load = util.load(AllDoNotTouchTable.class, attr.get(KEY_NAME).getS());
+      assertEquals(load.getKey(), attr.get(KEY_NAME).getS());
+      assertEquals(load.getStringAttribute(), attr.get(STRING_ATTRIBUTE).getS());
+    }
+  }
+
+  @Test(expectedExceptions = DynamoDBMappingException.class)
+  public void testLoadWithBadMissingSignatureFields() {
+    TestDynamoDBMapperFactory.createDynamoDBMapper(dynamo)
+        .load(StringAttributeTestClass.class, attrs.get(0).get(KEY_NAME).getS());
+  }
+
+  @DynamoDBTable(tableName = "aws-java-sdk-util-crypto")
+  public static final class AllDoNotTouchTable {
```

Comment:
Instead of testing the error case with `StringAttributeTestClass`, can we define a new class that extends and overrides this one to change `StringAttribute` from being ignored to being encrypted? Then the above tests will be less brittle in case `StringAttributeTestClass` is updated.

Reply:
make sense - Done